libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/NavigationRerouteController.kt

```diff
@@ -13,6 +13,8 @@ import com.mapbox.navigation.core.trip.session.RouteProgressObserver
  * Reroute controller allows changing the reroute logic externally. Use [MapboxNavigation.rerouteController]
  * to replace it.
  */
+@Deprecated("Custom rerouting logic is deprecated. " +
```

Comment:
As I know reroute controller is also used for observing reroute state. We probably shouldn't deprecate methods to get reroute state if we don't expose alternative API. What do you think?

Reply:
For the user to observe reroute states? Why would they need it? I thought it was for us to invoke interrupt when needed based on state (we check the state in `setRerouteController`).
But maybe you're right. We also provide a way to observer route refresh state, so yeah: no alternative - no deprecation.

Reply:
> For the user to observe reroute states? Why would they need it? 

To show loading indicator when user deviates from a route to let them know that we're calculating a new route.